File: client-panel/frontend/src/app/_models/email.ts
import {Campaign} from "./campaign";

export class Email {
  clientID: number;
  fromEmailAddress: string;
  toEmailAddresses: string[];
  ccEmailAddresses: string[];
  bccEmailAddresses: string[];
  replyToEmailAddresses: string[];
  emailSubject: string;
  emailBody: string;
  userID: string;
}

export class EmailTemplate {
  id: number;
  clientID: number;
  appuserID: number;
  name: string;
  emailTemplateBody: string;
  emailTemplateSubject: string;
  parentID: number;
  from: string;
  messageType: MessageType;
  tags: string;
}

export enum MessageType{
  TRANSACTIONAL,
  PROMOTIONAL
}

export class EmailCampaign {
  emailCampaignId: number;
  emailCampaignClientID: number;
  appuserID: number;
  campaign: Campaign;
  emailTemplateID: number;
}

File: client-panel/src/main/kotlin/com/und/model/Email.kt
package com.und.model

import com.und.model.mongo.eventapi.EventUser
import javax.mail.internet.InternetAddress

data class Email(
        var clientID: Long,
        var fromEmailAddress: InternetAddress? = null,
        var toEmailAddresses: Array<InternetAddress>,
        var ccEmailAddresses: Array<InternetAddress>? = null,
        var bccEmailAddresses: Array<InternetAddress>? = null,
        var replyToEmailAddresses: Array<InternetAddress>? = null,
        var emailSubject: String? = null,
        var emailBody: String? = null,
        var emailTemplateId: Long,
        var emailTemplateName: String,
        var data: MutableMap<String, Any> = mutableMapOf(),
        var eventUser: EventUser? = null
)